src/provider/Provider.ts

```diff
@@ -245,6 +246,38 @@ export class Provider {
     }
   }
 
+  /** Get Compute Environments
+   * @return {Promise<ComputeEnvironment[]>} urlDetails
+   */
+  public async getComputeEnvironments(
+    providerUri: string,
+    signal?: AbortSignal
+  ): Promise<ComputeEnvironment[]> {
+    const providerEndpoints = await this.getEndpoints(providerUri)
+    const serviceEndpoints = await this.getServiceEndpoints(
+      providerUri,
+      providerEndpoints
+    )
+    const envs: ComputeEnvironment[] = []
+    const path = this.getEndpointURL(serviceEndpoints, 'computeEnvironments')
+      ? this.getEndpointURL(serviceEndpoints, 'computeEnvironments').urlPath
+      : null
+    if (!path) return null
+    try {
+      const response = await fetch(path, {
+        method: 'GET',
+        headers: {
+          'Content-Type': 'application/json'
+        },
+        signal: signal
+      })
+      const envs: ComputeEnvironment[] = await response.json()
+      return envs
+    } catch (e) {
+      return null
```

Comment:
seeing this a lot in ocean.js but what's wrong with logging errors? `LoggerInstance.error(e.message)` should be everywhere